cmd/ocm/create/cluster/cmd.go

```diff
@@ -180,76 +182,129 @@ func run(cmd *cobra.Command, argv []string) error {
+		err = arguments.PromptString(fs, "aws-access-key-id")
 		if err != nil {
-			return fmt.Errorf("unable to retrieve versions: %s", err)
+			return err
 		}
 
-		clusterVersion = c.DropOpenshiftVPrefix(args.version)
-		if !sets.NewString(versionList...).Has(clusterVersion) {
-			return fmt.Errorf(
-				"A valid version number must be specified\n"+
-					"Valid versions: %+v\n"+
-					"Current default version is %v",
-				versionList, defaultVersion)
+		err = arguments.PromptPassword(fs, "aws-secret-access-key")
+		if err != nil {
+			return err
 		}
-		clusterVersion = c.EnsureOpenshiftVPrefix(clusterVersion)
+	}
+	err = arguments.CheckIgnoredCCSFlags(args.ccs)
+	if err != nil {
+		return err
+	}
+
+	regionSet := sets.NewString()
+	regions, err := provider.GetRegions(connection.ClustersMgmt().V1(), args.provider, args.ccs)
+	if err != nil {
+		return err
+	}
+	for _, region := range regions {
+		regionSet.Insert(region.ID())
+	}
+	err = arguments.PromptOrCheckOneOf(fs, "region", regionSet.List())
+	if err != nil {
+		return err
+	}
+
+	if args.region == "us-east-1" && args.provider != "aws" {
+		return fmt.Errorf("if specifying a non-aws cloud provider, region must be set to a valid region")
+	}
+
+	expiration, err := c.ValidateClusterExpiration(args.expirationTime, args.expirationSeconds)
+	if err != nil {
+		return fmt.Errorf(fmt.Sprintf("%s", err))
+	}
+
+	// Check and set the cluster version
+	versionList, defaultVersion, err := c.GetEnabledVersions(cmv1Client)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve versions: %s", err)
 	}
 
+	if args.version == "" {
+		args.version = defaultVersion
+	}
+	args.version = c.DropOpenshiftVPrefix(args.version)
+	err = arguments.PromptOrCheckOneOf(fs, "version", versionList)
+	if err != nil {
+		return err
+	}
+	clusterVersion := c.EnsureOpenshiftVPrefix(args.version)
+
 	// Retrieve valid/default flavours
-	flavourList := sets.NewString()
+	flavourSet := sets.NewString()
 	flavours, err := fetchFlavours(cmv1Client)
 	if err != nil {
 		return fmt.Errorf("unable to retrieve flavours: %s", err)
 	}
 	for _, flavour := range flavours {
-		flavourList.Insert(flavour.ID())
+		flavourSet.Insert(flavour.ID())
 	}
 
-	// Check and set the cluster flavour
-	var clusterFlavour string
-	if args.flavour != "" {
-
-		if !flavourList.Has(args.flavour) {
-			return fmt.Errorf("A valid flavour number must be specified\nValid flavours: %+v", flavourList.List())
-		}
-		clusterFlavour = args.flavour
+	err = arguments.CheckOneOf(fs, "flavour", flavourSet.List())
+	if err != nil {
+		return err
 	}
 
 	if args.private && args.provider != "aws" {
 		return fmt.Errorf("Setting cluster as private is not supported for cloud provider '%s'", args.provider)
 	}
 
 	// Compute node instance type:
-	computeMachineType := args.computeMachineType
-
-	computeMachineType, err = c.ValidateMachineType(cmv1Client, args.provider, computeMachineType)
+	machineTypeList, err := provider.GetMachineTypeIDs(cmv1Client, args.provider)
```

Comment:
can you split this change to another PR?

Reply:
Gladly! #198